services/templeton/src/messaging/amqp.go

```diff
@@ -0,0 +1,216 @@
+package messaging
+
+import (
+	"logcabin"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/streadway/amqp"
+)
+
+var logger = logcabin.New()
+
+var (
+	//ReindexExchange is the name of the exchange that full-reindex info is passed around on.
+	ReindexExchange = "de"
+
+	//ReindexAllKey is the routing/binding key for full reindex messages.
+	ReindexAllKey = "index.all"
+
+	//ReindexTemplatesKey is the routing/binding key for templates reindex messages.
+	ReindexTemplatesKey = "index.templates"
+)
+
+// MessageHandler defines a type for amqp.Delivery handlers.
+type MessageHandler func(amqp.Delivery)
+
+type aggregationMessage struct {
+	handler  MessageHandler
+	delivery *amqp.Delivery
+}
+
+type consumer struct {
+	exchange     string
+	exchangeType string
```

Comment:
Allowing the exchange type to be configured here is a modification in this service, since the `de` exchange used for the full-reindexing messages is a `direct` exchange, not a `topic` exchange.